app/src/main/java/ch/hikemate/app/model/authentication/FirebaseAuthRepository.kt

```diff
@@ -33,9 +39,10 @@ class FirebaseAuthRepository {
       context: Context,
       coroutineScope: CoroutineScope,
       credentialManager: CredentialManager = CredentialManager.create(context),
+      startAddAccountIntentLauncher: ManagedActivityResultLauncher<Intent, ActivityResult>?,
```

Comment:
Question: I haven't the code for the View, but is it necessary to pass the ActivityLauncher from the View to the Model? Isn't there any way to get it in the ViewModel or the Model from the context?

Reply:
I wrote a possible implementation of the launcher while testing my functionality, and it looked something like this:
```kotlin 
val addAccountLauncher =
          rememberLauncherForActivityResult(
            contract = ActivityResultContracts.StartActivityForResult()) {
              result: ActivityResult ->
            authViewModel.signInWithGoogle(coroutineScope, context, null)
            Toast.makeText(context, "success", Toast.LENGTH_SHORT).show()
            Log.d("MainActivity", "addAccountLauncher result: $result")
          }
```

This `rememberLauncherForActivityResult` is only callable from a composable, so in the UI. This is why I decided to pass it all the way down... If you know a nicer way to solve this I'd be happy to hear, I also don't love this solution.

Reply:
Oh okay thanks!